mitodl/odl-video-service
# Transcoding Videos

#### What are the relevant tickets?
closes https://github.com/mitodl/odl-video-service/issues/19
closes https://github.com/mitodl/odl-video-service/issues/3

[The django-elastictranscoder PR](https://github.com/mitodl/django-elastic-transcoder/pull/1) should probably be merged first.

#### What's this PR do?
- Based on preset ID's in settings, an AWS ElasticTranscode job will be created and run immediately after a video is uploaded (and runs asynchronously), generating an HLS playlist and thumbnails.  The current default presets are 2M, 1M, and 600K HLS video, but we should probably create our own higher-resolution presets.

- When a video detail page is loaded, the transcode job's status for that video is checked (TODO: make use of SNS notifications to trigger this).  If the job is complete, a VideoFile model object is created for the playlist, and VideoThumbnail objects are created for each thumbnail (1 or more depending on length of video).

- When a video is deleted, all the associated VideoFile and VideoThumbnail objects along with associated S3 objects will be deleted.

#### How should this be manually tested?
You will need AWS credentials, Dropbox application credentials, etc as described in the README to run this application.  Once running, upload a video, then click 'view your uploaded videos', and click the link to your uploaded video.  This will load the video detail page.  At first you will probably just see links to the 'Original' video (the one uploaded).  If you refresh the page after a few minutes, you should see links for an 'HLS' video and one or more thumbnails.
